src/harm.cpp

```diff
@@ -209,4 +210,47 @@ int Harm::AdjustHarmGrpsSpacing(FunctorParams *functorParams)
     return FUNCTOR_SIBLINGS;
 }
 
+int Harm::Transpose(FunctorParams *functorParams)
+{
+    TransposeParams *params = dynamic_cast<TransposeParams *>(functorParams);
+    assert(params);
+
+    LogDebug("Transposing harm");
+
+    Text *textObject = dynamic_cast<Text *>(this->GetChild(0, TEXT));
+    std::wstring text = textObject->GetText();
```

Comment:
Always `assert` the result of a dynamic _cast when something should be return. In cases where it is not sure (such as this one), handle it properly.

Reply:
Does anything more need to be done besides `assert(textObject)`?

Reply:
Sorry, I was not clear. In this case, you need to exit if it is NULL. This should not happen, but it can. If `<harm>` is empty, for whatever reason, there will be no text child. (`assert` should be used to check that nothing went wrong but you expect this never to happen.)

Reply:
Okay, I think I've fixed this. Though it's crashing elsewhere (without transposing) when given an empty harm:

gdb says

```
Thread 2 received signal SIGSEGV, Segmentation fault.
0x0000000100074bd1 in vrv::FloatingCurvePositioner::FloatingCurvePositioner (
    this=<optimized out>, object=<optimized out>, alignment=<optimized out>,
    spanningType=<optimized out>)
    at verovio/src/floatingobject.cpp:379
379	}
```

(But I don't really see what the problem is there...)